src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -198,6 +205,34 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
             bool hasParameterBindingAttribute = false;
             Exception invalidInvokeBindingException = null;
 
+            bool triggerHasReturnBinding = false;
+            if (TypeUtility.TryGetReturnType(method, out Type methodReturnType))
+            {
+                if (bindingDataContract != null && bindingDataContract.TryGetValue(ReturnParamName, out Type triggerReturnType))
+                {
+                    // CONSIDER: Special casing for JObject and related types for cross-language support?
+                    if (!triggerReturnType.IsAssignableFrom(methodReturnType))
```

Comment:
>if (!triggerReturnType.IsAssignableFrom(methodReturnType)) [](start = 20, length = 58)

We probably need to go through the converter managers here, which  would handle this.   For now, the trigger can just provide a contract of type  Object, which makes this moot.